plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -256,39 +256,53 @@ struct block_time_tracker {
       block_idle_time += idle;
    }
 
-   void add_fail_time( const fc::microseconds& fail_time ) {
-      trx_fail_time += fail_time;
-      ++trx_fail_num;
+   void add_fail_time( const fc::microseconds& fail_time, bool is_transient ) {
+      if( is_transient ) {
+         // transient time includes both success and fail time
+         transient_trx_time += fail_time;
+         ++transient_trx_num;
+      } else {
+         trx_fail_time += fail_time;
+         ++trx_fail_num;
+      }
    }
 
-   void add_success_time( const fc::microseconds& time ) {
-      trx_success_time += time;
-      ++trx_success_num;
+   void add_success_time( const fc::microseconds& time, bool is_transient ) {
+      if( is_transient ) {
+         transient_trx_time += time;
+         ++transient_trx_num;
+      } else {
+         trx_success_time += time;
+         ++trx_success_num;
+      }
```

Comment:
These are awkward. I would do something like:

```
struct block_time_tracker {
   enum class trx { success, fail, transient, last };

   void add_trx_time( const fc::microseconds& fail_time, trx trx_type ) {
         processing_time[(int)trx_type] += fail_time;
         ++num_processed[(int)trx_type];
   }
   
   uint32_t num_processed[(int)trx::last];
   fc::microseconds processing_time[(int)trx::last];
};
```

and use like `_time_tracker.add_trx_time(end - start, trx::fail);`

Reply:
I call that way more awkward. Looks like C code.

Reply:
> I call that way more awkward. 

I guess we will have to agree to disagree. 

> Looks like C code.

What exactly? The casts?

Reply:
The casts, the c-array, using the last enum value for the size of the c-array, and using an array for what in my opinion is better expressed as separate variables.

It is certainly more compact.